src/main/java/duke/DukeMethods.java

```diff
@@ -0,0 +1,175 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.ToDos;
+import duke.text.Text;
+
+import java.util.ArrayList;
+
+public class DukeMethods {
+
+    public static void printIntroduction() {
+        System.out.println(Text.LINE);
+        System.out.print(Text.LOGO);
+        System.out.println(Text.LINE);
+        System.out.println(Text.HELLO_MESSAGE);
+        System.out.println(Text.LINE);
+    }
+
+    public static void printList(ArrayList<Task> taskList) {
+        int taskNumber = 1;
+        System.out.println(Text.LINE);
+        System.out.println(Text.LIST_MESSAGE);
+        for (Task t : taskList) {
+            System.out.println(taskNumber + "." + t);
+            taskNumber++;
+        }
+        System.out.println(Text.LINE);
+    }
+
+    public static void printHelpList() {
+        System.out.println(Text.LINE);
+        System.out.println(Text.HELP_LIST);
+        System.out.println(Text.LINE);
+    }
+
+    public static void markCompletedTask(String userCommand, ArrayList<Task> taskList) {
+        String[] taskToMark = userCommand.split(" ");
+        int taskNumberToMark;
+        try {
+            if (taskToMark.length <= 1) {
+                throw new DukeException();
```

Comment:
You could initialize a constructor in DukeException (pass this variable to constructor of Exception) which takes in an error message, then print the error message. That way you can do e.errorMessage() in the catch blocks.